atomix/cluster/src/main/java/io/atomix/primitive/proxy/impl/LogProxySession.java

```diff
@@ -441,7 +441,7 @@ public void publish(final PrimitiveEvent event) {
         final Method method = eventMethods.get(event.type());
         if (method != null) {
           try {
-            method.invoke(client, (Object[]) decode(event.value()));
+            method.invoke(client, (Object[]) super.decode(event.value()));
```

Comment:
So this will now call AbstractSession#decode instead of LogProxySession#decode, since `LocalSession` was a non-static inner class. I guess I don't understand the change here :thinking:

Reply:
The thing is it also called before the super but it was not explicit. If you inherit a method then you will call it before calling methods from outer scope. If you want to call `LocalProxySession#decode` you need to write. `LocalProxySession.this.decode()`